plugin/evm/atomic/atomictest/shared_memories.go

```diff
@@ -0,0 +1,85 @@
+package atomictest
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/chains/atomic"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/snowtest"
+	"github.com/stretchr/testify/assert"
+)
+
+type SharedMemories struct {
+	ThisChain   atomic.SharedMemory
+	PeerChain   atomic.SharedMemory
+	thisChainID ids.ID
+	peerChainID ids.ID
+}
+
+func (s *SharedMemories) AddItemsToBeRemovedToPeerChain(ops map[ids.ID]*atomic.Requests) error {
+	for _, reqs := range ops {
+		puts := make(map[ids.ID]*atomic.Requests)
+		puts[s.thisChainID] = &atomic.Requests{}
+		for _, key := range reqs.RemoveRequests {
+			val := []byte{0x1}
+			puts[s.thisChainID].PutRequests = append(puts[s.thisChainID].PutRequests, &atomic.Element{Key: key, Value: val})
+		}
+		if err := s.PeerChain.Apply(puts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *SharedMemories) AssertOpsApplied(t *testing.T, ops map[ids.ID]*atomic.Requests) {
+	t.Helper()
+	for _, reqs := range ops {
+		// should be able to get put requests
+		for _, elem := range reqs.PutRequests {
+			val, err := s.PeerChain.Get(s.thisChainID, [][]byte{elem.Key})
+			if err != nil {
+				t.Fatalf("error finding puts in peerChainMemory: %s", err)
+			}
```

Comment:
It feels a bit odd to mix explicit error checking with the usage of `assert`.
```suggestion
			require.NoError(t, err)
```